pkg/agent/agent.go

```diff
@@ -237,6 +250,15 @@ func (agent *Agent) run(ctx context.Context) {
 		}
 	}, time.Duration(0))
 
+	go wait.UntilWithContext(ctx, func(ctx context.Context) {
+		mcs.NewController(
```

Comment:
seems we could new this controller in `NewAgent`.

Reply:
Yes, that's semantics reasonable. 

But We need `parentDedicatedKubeConfig`, the field only availlable after agent register successfully. I think sockets.NewController is in the same case.

Reply:
please take another look @dixudx